src/components/me.js

```diff
@@ -83,11 +83,11 @@ class Me extends User {
         if (this.chatEngine.ceConfig.enableSync) {
 
             // subscribe to the events on our sync chat and forward them
-            this.sync.on('$.session.notify.chat.join', (payload) => {
+            this.sync.on('$.system.chat.join', (payload) => {
```

Comment:
Why exactly are these changing?

Reply:
Good question. This pull request implements a new event that fires when a user manually leaves the room. This is an internal leave event used by the system, so we call it ```$.system.leave```.  https://github.com/pubnub/chat-engine/pull/297/files/7de049468754612007d2f0a38dd6bf91ba12a214#diff-b068c43baad69f1bda17466ce044da2bR486

The ```$.session.notify``` events are also "system" (private) events that are not for the user. As I'm introducing a new naming convention just for manual leave events, I think it makes sense to switch over other internal events.

This does not affect public events and no migration is needed.

In the future, it is possible to get rid of all internal system events through an internal plugin system that augments payloads. I may spend some time playing around with this today.

Reply:
Cool. That certainly follows I saw you add the new leave. I guess my confusion is I see you adding the others so I'm not totally sure how you can listen for these events without changing the names when they are fired? I think this was an oversight on my part I see that two of the places you changed are the ```on``` and the are two places are the ```emit```. I think as long as these are the only places the new naming convention makes a lot of sense and this should be good to go!

Reply:
Yeah the two changes are the system emitting and listening for the same private event. So new versions will work no problem as the strings match.

Reply:
Playing around with moving this stuff into plugins now

Reply:
My attempt at refactor revealed that fixing this will be move involved.

Reply:
👍